### ISSUE TITLE
Evaluation of dense decoder

### ISSUE THREAD

**Issue by wwjwhen** [JianshuZhang/WAP#6]
Hi jianshu, the Dense's ExpRate on CROHME 2014 in your Multi-Scale paper is reported to be 50.1%. My question is whether the  result is obtained on CROHME 2014 test set i.e. the images in data\off_image_test\ folder?

**Comment by JianshuZhang** [comment]
Yes, the report result is obtained on CROHME2014, images in data/off_image_test

**Comment by JianshuZhang** [comment]
Sorry for the delay, my github email address is disabled for several days. The published results used an ensemble of 5 models, the ensemble codes will be released in soon future, waiting for the notification of another paper's submission.

**Comment by wwjwhen** [comment]
OK , thanks! And looking forward for that! So the result is the best ensemble validation result during training?

**Comment by JianshuZhang** [comment]
Yes



xysszjs@mail.ustc.edu.cn
 
From: Wenji Wang
Date: 2018-05-07 11:06
To: JianshuZhang/WAP
CC: Jianshu Zhang; Comment
Subject: Re: [JianshuZhang/WAP] Evaluation of dense decoder (#6)
OK , thanks! And looking forward for that! So the result is the best ensemble validation result during training?

**Comment by wwjwhen** [comment]
Hi, jianshu. In your **WAP** paper you mention that you use **CROHME2013 test set** as validation set during training, and your reported results on competition test set are fairly comparable with others. So there seems two ways to get 50.1% result in your **Dense** paper:
1 Training with 8836 math expressions and validate the model with **CROHME2013 test set** during training to get the best model then run **CROHME2014 test set** as test set with the best saved model and the testing result is 50.1% after ensemble.
2 Training with 8836 math expressions and validate the model with **CROHME2014 test set** during training on ensemble model and the 50.1% result is the best validation result obtained during the training process.
According to your codes, I think No.2 tends to be the correct, so which one was used in your experiments?

**Comment by JianshuZhang** [comment]
During my experiments, I used CROHME2013 for validation (as you said 1). While there is trivial difference between using CROHME2013 or CROHME2014 as validation set, they almost choose the same best training model. The released data only contains CROHME2014 training and testing set. The published results are based on CROHME2014 test set.



xysszjs@mail.ustc.edu.cn
 
From: Wenji Wang
Date: 2018-05-08 09:27
To: JianshuZhang/WAP
CC: Jianshu Zhang; Comment
Subject: Re: [JianshuZhang/WAP] Evaluation of dense decoder (#6)
Hi, jianshu. In your WAP paper you mention that you use CROHME2013 test set as validation set during training, and your reported results on competition test set are fairly comparable with others. So there seems two ways to get 50.1% result in your Dense paper:
1 Training with 8836 math expressions and validate the model with CROHME2013 test set during training to get the best model then run CROHME2014 test set as test set with the best saved model and the testing result is 50.1% after ensemble.
2 Training with 8836 math expressions and validate the model with CROHME2014 test set during training on ensemble model and the 50.1% result is the best validation result obtained during the training process.
According to your codes, I think No.2 tends to be the correct, so which one was used in your experiments?

**Comment by wwjwhen** [comment]
So the published result 50.1% is obtained on CROHME2014 test set with the best model evaluated by CROHME2013 test set?

**Comment by JianshuZhang** [comment]
Yes



xysszjs@mail.ustc.edu.cn
 
From: Wenji Wang
Date: 2018-05-08 10:17
To: JianshuZhang/WAP
CC: Jianshu Zhang; Comment
Subject: Re: [JianshuZhang/WAP] Evaluation of dense decoder (#6)
So the published result 50.1% is obtained on CROHME2014 test set with the best model evaluated by CROHME2013 test set?

**Comment by wwjwhen** [comment]
Another  question, when I am using CROHME2013 test set as validation set, I find that the ExpRate  on validation set is far lower than that on CROHME2014 test set, so when you did your experiments, what was the ExpRate on your CROHME2013 test set?

**Comment by JianshuZhang** [comment]
the CROHME2013 test set is harder, but the difference is not so far, nearly 4% Exprate lower than CROHME2014.   The latex notation in CROHME2013 inkml files need to be normalized.

my iPhone

**Comment by wwjwhen** [comment]
I find that some symbols in CROHME2013 latex notation are not included in dictionary.txt, so I replace them with visual-similar symbols in dictionary.txt, so do you mean this by "normalized" ?

**Comment by JianshuZhang** [comment]
Sorry for the delay, we were too busy for a OCR competition recently. The "normalization" not only represents the replacement of some visual-similar symbols, the "normalization" represents the latex caption of one formula must be unique. For example, x _ 2 ^ 3 and x ^ 3 _ 2 both denotes the same formula, but they are different representations, such cases must be normalized.

**Comment by wwjwhen** [comment]
Well, I see, thx!
